File: easy/1. Two Sum/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int length = nums.length;
        var map = new HashMap<Integer, Integer>();
        for (int i = 0; i < length; i++) {
            map.put(nums[i], i);
        }

        for (int i = 0; i < length; i++) {
            int num = nums[i];
            int remainder = target - num;
            if(map.containsKey(remainder) && map.get(remainder) != i) {
                return new int[]{i, map.get(remainder)};
            }
        }

        return new int[]{};
    }
}

File: easy/1. Two Sum/Solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val length = nums.size
        val map = HashMap<Int, Int>()

        for (i in 0 until length) {
            map[nums[i]] = i
        }

        for (i in 0 until length) {
            val num = nums[i]
            val remainder = target - num
            if (map.containsKey(remainder) && map[remainder] != i) {
                return intArrayOf(i, map[remainder]!!)
            }
        }

        return intArrayOf()
    }
}